src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -47,18 +42,27 @@ public EventPublishingController(final EventPublisher publisher, final EventType
     public ResponseEntity postEvent(@PathVariable final String eventTypeName, @RequestBody final String eventsAsString,
                                     final NativeWebRequest nativeWebRequest) {
         LOG.trace("Received event {} for event type {}", eventsAsString, eventTypeName);
+        final EventTypeMetrics eventTypeMetrics = eventTypeMetricRegistry.metricsFor(eventTypeName);
 
         try {
-            final EventTypeMetrics eventTypeMetrics = eventTypeMetricRegistry.metricsFor(eventTypeName);
+            final ResponseEntity response = postEventInternal(eventTypeName, eventsAsString, nativeWebRequest, eventTypeMetrics);
+            eventTypeMetrics.incrementResponseCount(response.getStatusCode().value());
+            return response;
+        } catch (RuntimeException ex) {
+            eventTypeMetrics.incrementResponseCount(-1); // yep, dunno response code here, related to event type.
```

Comment:
Yes, I think Spring turns every uncaught exception into 500. I think it would be correct to consider that here we have 500 response code.